docs/src/manuals/meta-specification.md

```diff
@@ -57,35 +63,185 @@ List of available options:
+using RxInfer
+
+#create data
+y_data = 4.0 
+
+#make model
+@model function gaussian_model()
+    y = datavar(Float64)
+
+    x ~ NormalMeanVariance(2.5, 0.5)
+    y ~ NormalMeanVariance(2*x, 2.)
+end
+
+#do inference
+inference_result = inference(
+    model = gaussian_model(),
+    data = (y = y_data,)
+)
+```
+However, let's say we would like to define a new inference procedure by introducing a meta structure to the `Normal` node that always yields a Normal distribution with mean $m$ for the outbound messages of the node. This is done as follows:
+```julia
+#create your new meta structure for Normal node
+struct MetaNormal
+    m::Real
+end
+
+#function that gets the parameter in meta
+getmean(meta::MetaNormal) = meta.m
+
+#define rules with meta for the Normal node
+@rule NormalMeanVariance(:out, Marginalisation) (q_μ::Any, q_v::PointMass, meta::MetaNormal) = begin
+    return NormalMeanVariance(getmean(meta), mean(q_v))
+end
+
+@rule NormalMeanVariance(:μ, Marginalisation) (q_out::Any, q_v::PointMass, meta::MetaNormal) = begin
+    return NormalMeanVarince(getmean(meta), mean(q_v))
+end
+
+#make model
+@model function gaussian_model_with_meta()
+    y = datavar(Float64)
+
+    x ~ NormalMeanVariance(2.5, 0.5)
+    y ~ NormalMeanVariance(2*x, 2.)
+end
+
+#specify meta for the model
+@meta function model_meta(m)
+    NormalMeanVariance(y,x) -> MetaNormal(m)
+end
+
+my_meta = model_meta(3) # mean = 3
+
+#do inference
+inference_result = inference(
+    model = gaussian_model(),
+    data = (y = y_data,),
+    meta = my_meta
+)
+```
+
+**Disclaimer**: The above example is not mathematically correct. It is only used to show how we can work with `@meta` as well as how to create a meta structure for a node in `RxInfer.jl`.
+
+## Nodes with meta data in RxInfer
+
+|    Node                                 |   Meta                                                                   |
+| :-------------------------------------- | :----------------------------------------------------------------------- |
+| AR                                      | ARmeta                                                                   |
+| GCV                                     | GCVMetadata                                                              |
+| [DeltaFn](@id delta-node-manual)        | DeltaMeta                                                                |
+| Probit                                  | Union{Nothing, ProbitMeta}                                               |
+| BIFM                                    | BIFMMeta                                                                 |
+| Flow                                    | FlowMeta                                                                 |
+| dot                                     | Union{Nothing, MatrixCorrectionTools.AbstractCorrectionStrategy}         |
+| (*)                                     | Union{Nothing, MatrixCorrectionTools.AbstractCorrectionStrategy}         |
```

Comment:
We should generate this table automatically [see](https://biaslab.github.io/ReactiveMP.jl/stable/lib/rules/rules/#lib-rules-table), or at least it should be clickable.

Reply:
I would not include the table in this form, but mention to refer to individual node's documentation to see if they use some meta objects. It would be difficult to maintain this table by hand, automatically generated is also fine.